x-pack/plugins/entity_manager/server/lib/entity_client.ts

```diff
@@ -170,4 +170,72 @@ export class EntityClient {
     this.options.logger.info(`Stopping transforms for definition [${definition.id}]`);
     return stopTransforms(this.options.esClient, definition, this.options.logger);
   }
+
+  async getEntitySources({ type }: { type: string }) {
+    const result = await this.options.esClient.search<EntitySource>({
+      index: 'kibana_entity_definitions',
+      query: {
+        bool: {
+          must: {
+            term: { entity_type: type },
+          },
+        },
+      },
+    });
+
+    return result.hits.hits.map((hit) => hit._source) as EntitySource[];
+  }
+
+  async searchEntities({
+    sources,
+    start,
+    end,
+    limit = 10,
+  }: {
+    sources: EntitySource[];
+    start: string;
+    end: string;
+    limit?: number;
+  }) {
+    const entities = await Promise.all(
+      sources.map(async (source) => {
+        const mandatoryFields = [source.timestamp_field, ...source.identity_fields];
+        const { fields } = await this.options.esClient.fieldCaps({
+          index: source.index_patterns,
+          fields: [...mandatoryFields, ...source.metadata_fields],
+        });
+
+        const sourceHasMandatoryFields = mandatoryFields.every((field) => !!fields[field]);
+        if (!sourceHasMandatoryFields) {
+          // we can't build entities without id fields so we ignore the source.
+          // filters should likely behave similarly.
+          this.options.logger.info(
+            `Ignoring source for type [${source.type}] with index_patterns [${source.index_patterns}] because some mandatory fields [${mandatoryFields}] are not mapped`
+          );
+          return [];
+        }
+
+        // but metadata field not being available is fine
+        const availableMetadataFields = source.metadata_fields.filter((field) => fields[field]);
```

Comment:
And this is the thing that avoids the ES|QL behaviour of failing if a field is not mapped?